linetools/isgm/abssystem.py

```diff
@@ -0,0 +1,202 @@
+""" Class for  absorption systems
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+# Python 2 & 3 compatibility
+try:
+    basestring
+except NameError:
+    basestring = str
+
+import numpy as np
+import warnings
+from abc import ABCMeta
+
+from astropy import units as u
+from astropy.table import QTable
+from astropy import constants as const
+from astropy.coordinates import SkyCoord
+
+from linetools.isgm.abscomponent import AbsComponent
+
+class AbsSystem(object):
+    """
+    Class for an absorption line system
+
+    Parameters
+    ----------
+    abs_type : str
```

Comment:
As I said before, I'm not sure about this 'abs_type' (at least being mandatory).